From: Miroslav Benes <mbenes@suse.cz>
Date: Tue, 5 Mar 2019 13:55:14 +0100
Subject: livepatch: Define a macro for new API identification
Patch-mainline: Never, SUSE-specific
References: bsc#1071995 fate#323487

Register/unregister stage was removed from livepatch. Define
KLP_NOREG_API macro, so that livepatches (kernel modules) can easily
identify if the kernel provides the new API or the old one with the
registration step.

Signed-off-by: Miroslav Benes <mbenes@suse.cz>
---
 include/linux/livepatch.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/livepatch.h b/include/linux/livepatch.h
index 53551f470722..caad8e3cd55f 100644
--- a/include/linux/livepatch.h
+++ b/include/linux/livepatch.h
@@ -35,6 +35,8 @@
 #define KLP_UNPATCHED	 0
 #define KLP_PATCHED	 1
 
+#define KLP_NOREG_API
+
 /**
  * struct klp_func - function structure for live patching
  * @old_name:	name of the function to be patched

